### PR TITLE
NSB.Metrics bump up to 1.1

### PR DESCRIPTION
**Pull request by Scooletz**
PR bumping up `NSB.Metrics` to `1.1.0` Includes:

- smoke tested samples
- reporting split into two partials: 1.0 with trace/log support and 1.1 with custom observers
- tracing sample version limited to 1.0.*
- observers for 1.1 snippet added

For now, it's based on `1.1.0-beta0001` published on MyGet.

### CODE REVIEW

File: Snippets/Metrics/Metrics_1.1/Configuration.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        {
+            #region Metrics-Enable
+
+            var metricsOptions = endpointConfiguration.EnableMetrics();
+
+            #endregion
+
+            #region Metrics-Observers
+
+            metricsOptions.RegisterObservers(ctx => { });
```
Comment (Scooletz) [inline Snippets/Metrics/Metrics_1.1/Configuration.cs]: What would be the best thing to show in here?

**Comment by Scooletz** [comment]
Ping @Particular/metrics-maintainers could you take a look at it?

File: samples/metrics/tracing/sample.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 ---
 title: Sending metrics data to trace log
 component: Metrics
-reviewed: 2017-04-04
+versions: '[1.0,1.1)'
```
Comment (SimonCropp) [inline samples/metrics/tracing/sample.md]: i dont think a range makes sense on a sample

File: Snippets/Metrics/Metrics_1.1/Metrics_1.1.csproj [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net462</TargetFramework>
+    <RootNamespace>Metrics_1_1</RootNamespace>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="6.2.1" />
```
Comment (SimonCropp) [inline Snippets/Metrics/Metrics_1.1/Metrics_1.1.csproj]: u can prob drop this ref and let it be infered

**Comment by SimonCropp** [comment]
i think `NServiceBus.Metrics component` can generally be shortened to `Metrics component`

**Comment by SimonCropp** [comment]
@Scooletz just an FYI it is fine to merge this (while still only on myget) and include a prerelease marker https://github.com/Particular/docs.particular.net#pre-release-marker-file and come back and clean up the marker later

that way u get links to rendered doco for people to review/ share. also means less likelihood of conflicts due to a long running PR.

File: Snippets/Metrics/Metrics_1.1/Configuration.cs (hunk 1 above)
Comment (SimonCropp) [inline Snippets/Metrics/Metrics_1.1/Configuration.cs]: just write to the console for now?

Comment (tmasternak) [inline Snippets/Metrics/Metrics_1.1/Configuration.cs]: > just write to the console for now?

:thumbsup:

File: nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+## Reporting metrics data
```
Comment (tmasternak) [inline nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md]: Do we want to add note somewhere here about the fact that those methods are obsoleted and will be gone in 2.0?

File: samples/metrics/tracing/sample.md (hunk 2 above)
Comment (Scooletz) [inline samples/metrics/tracing/sample.md]: What about a case when we introduce a new API that we'd like encourage people to use? Shouldn't I use versions for it?

File: Snippets/Metrics/Metrics_1.1/Configuration.cs (hunk 1 above)
Comment (Scooletz) [inline Snippets/Metrics/Metrics_1.1/Configuration.cs]: Will do.

File: nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md (hunk 4 above)
Comment (Scooletz) [inline nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md]: The are obsoleted from 1.1 which is not described in these partial. Should I copy this to `[1.1,)` and mark as obsoleted? If yes, how should I mention the preferred way of reporting with observers? Mention it first, then say, that all the methods below are obsoleted now?

**Comment by Scooletz** [comment]
Thank you @SimonCropp for showing me the how to include a prerelease. I think that in this case, I'll leave it as it is. It won't have any merge conflicts and we're about to ship the Metrics anyway.

File: nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md (hunk 4 above)
Comment (Scooletz) [inline nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md]: I'll apply my comment.

File: samples/metrics/tracing/sample.md (hunk 2 above)
Comment (SimonCropp) [inline samples/metrics/tracing/sample.md]: in a doco page yes. in a sample u would just use the new api, no need for the versions header

File: Snippets/Metrics/Metrics_1/Configuration.cs [hunk 5]
```diff
@@ -44,6 +44,28 @@ public void ConfigureEndpoint(EndpointConfiguration endpointConfiguration)
                 interval: TimeSpan.FromSeconds(5));
 
             #endregion
+
+            #region Metrics-Observers [1.1,)
```
Comment (Scooletz) [inline Snippets/Metrics/Metrics_1/Configuration.cs]: @Particular/docs-maintainers How can I mark this snippet to get it properly displayed in `metrics_reporting_metrics_[1.1,).partial.md` ? Am I getting this versioning wrong?

File: samples/metrics/tracing/sample.md (hunk 2 above)
Comment (Scooletz) [inline samples/metrics/tracing/sample.md]: What's the "new API" @SimonCropp ? Any link?

File: Snippets/Metrics/Metrics_1/Configuration.cs (hunk 5 above)
Comment (SimonCropp) [inline Snippets/Metrics/Metrics_1/Configuration.cs]: partials are derived using the dirs. so to have a `metrics_reporting_metrics_[1.1,).partial.md` you would need a 1.1 here https://github.com/Particular/docs.particular.net/tree/metrics-1.1-bump-up/Snippets/Metrics

File: samples/metrics/tracing/sample.md (hunk 2 above)
Comment (SimonCropp) [inline samples/metrics/tracing/sample.md]: @Scooletz you referred to the "new api" in your previous comment

File: nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md (hunk 4 above)
Comment (Scooletz) [inline nservicebus/operations/metrics_reporting_metrics_[1.0,1.1).partial.md]: According to our discussion with @tmasternak I changed the doco, copying 1.0 to 1.1 and warn about getting obsolete.

File: Snippets/Metrics/Metrics_1/Configuration.cs (hunk 5 above)
Comment (Scooletz) [inline Snippets/Metrics/Metrics_1/Configuration.cs]: As I did before 😭 Will do.

**Comment by Scooletz** [comment]
Ping @Particular/docs-maintainers for a review

File: Snippets/Metrics/Metrics_1.1/Metrics_1.1.csproj [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net462</TargetFramework>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus.Metrics" Version="1.1.0-beta0001" />
```
Comment (SimonCropp) [inline Snippets/Metrics/Metrics_1.1/Metrics_1.1.csproj]: should this still be beta?

File: Snippets/Metrics/Metrics_1/Metrics_1.csproj [hunk 7]
```diff
@@ -4,7 +4,6 @@
     <TargetFramework>net462</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="NServiceBus" Version="6.2.1" />
-    <PackageReference Include="NServiceBus.Metrics" Version="1.0.0" />
+    <PackageReference Include="NServiceBus.Metrics" Version="1.1.0-beta0001" />
```
Comment (SimonCropp) [inline Snippets/Metrics/Metrics_1/Metrics_1.csproj]: this should be 1.0 still?

File: samples/performance-counters/PerfCounters_1/Sample/packages.config [hunk 8]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="NServiceBus" version="6.3.4" targetFramework="net461" />
-  <package id="NServiceBus.Metrics" version="1.0.1" targetFramework="net461" />
+  <package id="NServiceBus.Metrics" version="1.1.0-beta0001" targetFramework="net461" />
```
Comment (SimonCropp) [inline samples/performance-counters/PerfCounters_1/Sample/packages.config]: still beta?

File: samples/performance-counters/PerfCounters_1/Sample/Sample.csproj [hunk 9]
```diff
@@ -35,7 +35,7 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="NServiceBus.Metrics, Version=1.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
-      <HintPath>..\..\..\..\packages\NServiceBus.Metrics.1.0.1\lib\net452\NServiceBus.Metrics.dll</HintPath>
+      <HintPath>..\..\..\..\packages\NServiceBus.Metrics.1.1.0-beta0001\lib\net452\NServiceBus.Metrics.dll</HintPath>
```
Comment (SimonCropp) [inline samples/performance-counters/PerfCounters_1/Sample/Sample.csproj]: sill beta?

**Comment by Scooletz** [comment]
😊 It's no more beta @SimonCropp

File: Snippets/Metrics/Metrics_1/Metrics_1.csproj [hunk 10]
```diff
@@ -4,7 +4,6 @@
     <TargetFramework>net462</TargetFramework>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="NServiceBus" Version="6.2.1" />
-    <PackageReference Include="NServiceBus.Metrics" Version="1.0.0" />
+    <PackageReference Include="NServiceBus.Metrics" Version="1.1.0" />
```
Comment (SimonCropp) [inline Snippets/Metrics/Metrics_1/Metrics_1.csproj]: shouldn't this be 1.0?

Comment (Scooletz) [inline Snippets/Metrics/Metrics_1/Metrics_1.csproj]: This is a minor so it's ok to have it 1.1 as well (snippets versioned by project name/dir structure). Should I change it to 1.0?

**Comment by SimonCropp** [comment]
i will merge this to get the doco rendered. and do another PR for how i would propose to do the versioning

**Pull request merged**